product_pricelist_supplierinfo/models/product_product.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2018 Tecnativa - Vicent Cubells
+# Copyright 2018 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models, tools
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    def _get_supplierinfo_pricelist_price(self, rule, date=None,
+                                          quantity=None):
+        """Method for getting the price from supplier info."""
+        self.ensure_one()
+        domain = [('id', 'in', self.seller_ids.ids)]
```

Comment:
To get the value for `seller_ids.ids` the orm must do a first sql search query on the `product_supplier_info` table. To avoid this query, you should replace this first part of the domain by `['|', ('product_id', '=', self.id), ('product_tmpl_id', '=', self.product_tmpl_id)]`
An other way is to try to keep advantage of the cache by doing all the logic with the orm (no `search`)

Reply:
@lmignon accessing `ids` doesn't require to make the select under the hook. This is the only value that is already pre-stored when you make a browse or access a relational field. It's when you access other fields that prefetching machinery starts to work, make the select for the whole recordset and cache data for accessing. Can you please recheck?

I'm using this domain technique for readability, as using `filtered` or `if` conditions would affect it. The drawback is not too much or it's even better than the other way, as we only have 2 queries (search and select), reading one record (that it can even be on cache). On contrary, the other way you have to select all supplierinfo records, which might spend more time than the other 2 queries.

Reply:
@lmignon you're totally right about the extra query needed. What I was saying was for many2one indeed, not for many2many. Changed in next commit by the search. As this method is isolated and we don't know about cache status, let's go for the `search` and let ORM the rest.

Reply:
@pedrobaeza Thank you for the change. This part of the code looks better now. I'm still wonder if we could make the method `_get_supplierinfo_pricelist_price` works with a multi recordset. I'll comment on the method.

Reply:
What is the interest of making this multi-record aware? Having into account that `_compute_price_rule` is already for one record, it's difficult to handle this is a proper way I think, but I haven't tried.

Reply:
@pedrobaeza The main interest is that all the supplier info could be retrieved with one call to search for all the products of a same price list rule. 
You are right, the code could be a little bit more complex to read but could scale more efficiently. As I said, it's a details and you already have done a great work.  If I or someone else has a performance problem with this module, there will still be time to adapt it at that time. At this point, I think the bottleneck will first be the native odoo code before this one.